crates/router/src/connector/globalpay/transformers.rs

```diff
@@ -20,28 +23,45 @@ use crate::{
     types::{self, api, domain, storage::enums, transformers::ForeignTryFrom, ErrorResponse},
 };
 
+impl<T> From<(StringMinorUnit, T)> for GlobalPayRouterData<T> {
+    fn from((amount, item): (StringMinorUnit, T)) -> Self {
+        Self {
+            amount,
+            router_data: item,
+        }
+    }
+}
+
 type Error = error_stack::Report<errors::ConnectorError>;
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct GlobalPayMeta {
     account_name: Secret<String>,
 }
 
-impl TryFrom<&types::PaymentsAuthorizeRouterData> for GlobalpayPaymentsRequest {
+impl TryFrom<&GlobalPayRouterData<&types::PaymentsAuthorizeRouterData>>
+    for GlobalpayPaymentsRequest
+{
     type Error = Error;
-    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+    fn try_from(
+        item: &GlobalPayRouterData<&types::PaymentsAuthorizeRouterData>,
+    ) -> Result<Self, Self::Error> {
         let metadata: GlobalPayMeta =
-            utils::to_connector_meta_from_secret(item.connector_meta_data.clone())?;
+            utils::to_connector_meta_from_secret(item.router_data.connector_meta_data.clone())?;
         let account_name = metadata.account_name;
-        let (initiator, stored_credential, brand_reference) = get_mandate_details(item)?;
-        let payment_method_data = get_payment_method_data(item, brand_reference)?;
+        let (initiator, stored_credential, brand_reference) =
+            get_mandate_details(item.router_data)?;
+        let payment_method_data = get_payment_method_data(item.router_data, brand_reference)?;
         Ok(Self {
             account_name,
-            amount: Some(item.request.amount.to_string()),
-            currency: item.request.currency.to_string(),
-            reference: item.connector_request_reference_id.to_string(),
-            country: item.get_billing_country()?,
-            capture_mode: Some(requests::CaptureMode::from(item.request.capture_method)),
+            amount: Some(item.amount.to_owned()),
```

Comment:
can we do `let request_data = item.router_data`, Optional change!

Reply:
Not required bro